File: 1768-merge-strings-alternately/1768-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        ans = ""
        i, j = 0, 0
        while i < len(word1) and j < len(word2):
            ans += word1[i]
            i += 1
            ans += word2[j]
            j += 1
            
        if i != len(word1):
            for a in range(i, len(word1)):
                ans += word1[a]
                
        if j != len(word2):
            for a in range(j, len(word2)):
                ans += word2[a]
        
        return ans

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.rs
impl Solution {
    pub fn merge_alternately(word1: String, word2: String) -> String {
        let mut ans = String::new();
        let (mut i, mut j) = (0, 0);

        while i < word1.len() && j < word2.len() {
            ans.push(word1.chars().nth(i).unwrap());
            ans.push(word2.chars().nth(j).unwrap());
            i += 1;
            j += 1;
        }

        if i != word1.len() {
            ans.push_str(&word1[i..]);
        }

        if j != word2.len() {
            ans.push_str(&word2[j..]);
        }

        ans
    }
}
